asv_bench/benchmarks/utils.py

```diff
@@ -58,25 +58,47 @@ def gen_str_int_data(ncols, nrows, rand_low, rand_high):
             nrows, ncols, rand_low, rand_high
         )
     )
-    data = gen_int_data(ncols, nrows, rand_low, rand_high).copy()
+    data = gen_int_data(nrows, ncols, rand_low, rand_high).copy()
     data["gb_col"] = [
         "str_{}".format(random_state.randint(rand_low, rand_high)) for i in range(nrows)
     ]
     data_cache[cache_key] = weakdict(data)
     return data
 
 
-def gen_data(data_type, ncols, nrows, rand_low, rand_high):
+def gen_data(data_type, nrows, ncols, rand_low, rand_high):
     if data_type == "int":
-        return gen_int_data(ncols, nrows, rand_low, rand_high)
+        return gen_int_data(nrows, ncols, rand_low, rand_high)
     elif data_type == "str_int":
-        return gen_str_int_data(ncols, nrows, rand_low, rand_high)
+        return gen_str_int_data(nrows, ncols, rand_low, rand_high)
     else:
         assert False
 
 
-def generate_dataframe(impl, data_type, ncols, nrows, rand_low, rand_high):
-    cache_key = (impl, data_type, ncols, nrows, rand_low, rand_high)
+def generate_dataframe(
+    impl,
+    data_type,
+    nrows,
+    ncols,
+    rand_low,
+    rand_high,
+    groupby_ncols=None,
+    count_groups=None,
+):
+    if groupby_ncols and count_groups:
```

Comment:
We should either set a default value for `count_groups` or add this assert:
```python
assert not (
    groupby_ncols is None ^ count_groups is None
), "You must either specify both parameters 'groupby_ncols' and 'count_groups' or none of them."
```
otherwise, the user of this function may be confused when specify only the `groupby_ncols` parameter but see no effect on the output.